### PR TITLE
Refactor app and use WSGI to serve app in production

### PR DESCRIPTION
**Pull request by purelogiq**
## (This PR has been split into 4 commits to make it easy to review) 

Derived from #17. Please see the comments and original code made by @mars-f there.

Test with a docker-compose.override.yml that looks like:
```
version: '2'
services:
  lando-ui:
    build:
      context: ./
      dockerfile: ./docker/Dockerfile-prod
    ports:
      - "9000:9000"
    environment:
      - UWSGI_WORKERS=1
      - UWSGI_SOCKET=:9001
      - UWSGI_HTTP=:9000
      - VERSION_PATH=/app/version.json
      - SESSION_COOKIE_NAME=lando-ui.test:9000
      - SESSION_COOKIE_DOMAIN=lando-ui.test:9000
```

And make sure that you comment out the entire "volumes" section in the original docker-compose.yml file. As so:
```
#   volumes:
#      - ./landoui/:/app/landoui
#      - ./tests/:/app/tests
```
This is needed because the production docker image doesn't use docker-compose and copies over all the files inside the image, and so the volume section would override the files causing permission errors. 

### Changes from original:

- Changes the position of the assets folder to be compatible with both the production and dev environments (before production would either crash or serve the source static files which we don't want to do).

- Moves the installation of uwsgi from the `requirements.txt` file to a line in the production Dockerfile. This is because the dev Dockerfile was made much more complicated by the addition of the uwsgi and took a long time to build when there is no cache.

- Restructures the `create_app` method to be more like Lando API, which basically separates running the dev server from creating the app instance.

### CODE REVIEW

File: docker/Dockerfile-prod [hunk 1]
```diff
@@ -1,21 +1,65 @@
...
+    && apk add --update --no-cache --virtual .build-deps \
+           gcc \
+           libc-dev \
+           musl-dev \
+           linux-headers \
+           pcre-dev \
+           openssl-dev \
+           libffi-dev \
+    && pip install --no-cache -r /requirements.txt \
+    && pip install --no-cache uWSGI==2.0.15 \
```
Comment (mars-f) [inline docker/Dockerfile-prod]: uWSGI is now being installed without hashes, and all other deps are.  We should add the `--hash` line(s) and `--require-hashes` options here that were removed from requirements.txt.  If it's messy to add it directly to this Dockerfile, we could add a prod-requirements.txt file just for the uWSGI and psycopg2 stuff.

Comment (purelogiq) [inline docker/Dockerfile-prod]: I'll add it to the dockerfile directly, in the future a prod-requirements.txt might be very helpful.

Comment (mars-f) [inline docker/Dockerfile-prod]: Come to think of it, we have so many requirements files now, we might want to switch to a cleaner directory structure: add a `requirements/` directory, and put `dev.txt`, `common.txt`, `circle.txt`, and `production.txt` requirements files in there.

Comment (purelogiq) [inline docker/Dockerfile-prod]: I will do both in a new commit!

**Pull request merged**